constraint/pkg/client/client_test.go

```diff
@@ -487,6 +488,48 @@ func TestRemoveTemplate(t *testing.T) {
 	}
 }
 
+func TestTemplateCascadingDelete(t *testing.T) {
+	handler := &badHandler{Name: "h1", HasLib: true}
+	template := createTemplate(name("cascadingdelete"), crdNames("CascadingDelete"), targets("h1"))
+
+	d := local.New()
+	b, err := NewBackend(Driver(d))
+	if err != nil {
+		t.Fatalf("Could not create backend: %s", err)
+	}
+	c, err := b.NewClient(Targets(handler))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err = c.AddTemplate(context.Background(), template); err != nil {
+		t.Errorf("err = %v; want nil", err)
+	}
+
+	cst1 := newConstraint("CascadingDelete", "cascadingdelete", nil, nil)
+	if _, err = c.AddConstraint(context.Background(), cst1); err != nil {
+		t.Error("could not add first constraint")
+	}
+	cst2 := newConstraint("CascadingDelete", "cascadingdelete2", nil, nil)
+	if _, err = c.AddConstraint(context.Background(), cst2); err != nil {
+		t.Error("could not add second constraint")
+	}
+
+	if _, err = c.RemoveTemplate(context.Background(), template); err != nil {
+		t.Error("could not remove template")
+	}
+	if len(c.constraints) != 0 {
```

Comment:
Worth adding another template to show that cascading delete of one template leaves those intact?

Reply:
+1